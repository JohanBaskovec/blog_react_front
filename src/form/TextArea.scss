@import "./FormComponent.scss";
.TextArea {
  display: flex;
  resize: none;
  @extend .FormComponent;
}
