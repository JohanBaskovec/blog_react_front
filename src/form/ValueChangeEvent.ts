export class ValueChangeEvent<T> {
    constructor(private _name: string, private _value: T) {

    }

    get name(): string {
        return this._name;
    }

    get value(): T {
        return this._value;
    }
}

export type ValueChangeEventHandler<T> = (e: ValueChangeEvent<T>) => void;
