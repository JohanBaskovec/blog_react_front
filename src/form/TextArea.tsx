import React from "react";
import "./TextArea.scss";
import {FieldHelperProps, FieldInputProps, FieldMetaProps, useField} from "formik";

export interface TextAreaProps {
    style?: React.CSSProperties;
    name: string;
}

export function TextArea(props: TextAreaProps) {
    const [field]: [FieldInputProps<any>, FieldMetaProps<any>, FieldHelperProps<any>] = useField(props.name);
    // noinspection CheckTagEmptyBody
    return <textarea id={props.name}
                     className="TextArea"
                     {...field}
                     style={props.style}></textarea>
}
