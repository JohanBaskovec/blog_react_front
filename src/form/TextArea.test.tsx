import React from "react";
import TestRenderer from 'react-test-renderer';
import {fireEvent, render, screen} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import {ValueChangeEvent} from "./ValueChangeEvent";
import {TextArea} from "./TextArea";

describe('TextArea', () => {
    it("works", () => {
        let onChangeWorks = false;
        const onChange = (e: React.ChangeEvent) => {
            onChangeWorks = true;
        }
        let value = '';
        let onValueChangeEvent: ValueChangeEvent<string> | null = null;
        const onValueChange = (e: ValueChangeEvent<string>) => {
            value = e.value;
            onValueChangeEvent = e;
        }
        const name = 'inputName';
        const element = <TextArea name={name} value={value} onChange={onChange} onValueChange={onValueChange}/>;
        render(element);
        const input: HTMLElement = screen.getByRole('textbox');
        const text = 'test';
        fireEvent.input(input, {target: {value: text}});

        expect(onChangeWorks).toBe(true);
        expect(value).toBe(text);
        expect(onValueChangeEvent).not.toBeNull();
        expect(onValueChangeEvent!.name).toBe(name);
    });
    it("is rendered properly", () => {
        const tree = TestRenderer
            .create(<TextArea value='read-only' name='name'/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
