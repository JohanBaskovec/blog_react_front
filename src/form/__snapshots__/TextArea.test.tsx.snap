// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea is rendered properly 1`] = `
<textarea
  className="TextArea"
  id="name"
  onChange={[Function]}
  value="read-only"
/>
`;
